pkg/asset/cluster/tfvars.go

```diff
@@ -338,16 +339,29 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		}
 		preexistingnetwork := installConfig.Config.GCP.Network != ""
 
-		imageRaw, err := rhcospkg.GCPRaw(ctx, installConfig.Config.ControlPlane.Architecture)
+		archName := coreosarch.RpmArch(fmt.Sprintf("%s", installConfig.Config.ControlPlane.Architecture))
+		st, err := rhcospkg.FetchCoreOSBuild(ctx)
 		if err != nil {
-			return errors.Wrap(err, "failed to find Raw GCP image URL")
+			return err
+		}
+		starch, err := st.GetArchitecture(archName)
+		if err != nil {
+			return err
+		}
+
+		var gcpimage string
+		if starch.Images.Gcp != nil {
+			img := starch.Images.Gcp
+			gcpimage = fmt.Sprintf("projects/%s/global/images/%s", img.Project, img.Name)
+		} else {
+			return fmt.Errorf("%s: No GCP build found", st.FormatPrefix(archName))
```

Comment:
This error wrapping, there will be a lot of colons. Let's not add to that.
```suggestion
			return fmt.Errorf("No GCP build found for %s architecture", st.FormatPrefix(archName))
```

Reply:
Already taken care of by the `FormatPrefix` call: https://github.com/coreos/stream-metadata-go/blob/7c4d5d7d95b51e9fd2043375ee4b727f68c8f96a/stream/stream_utils.go#L7

Reply:
Ah yes, it would show up as 
`stream:rhcos-4.8 arch:x86_64: No GCP build found`
which I agree is ugly.  Hmm.  Maybe I'll change `FormatPrefix` to do instead `rhcos-4.8/x86_64:` ?

Reply:
https://github.com/coreos/stream-metadata-go/pull/17

Reply:
My concern is more the final resulting message that is presented to the user, not the isolated error here. The final error is going to look something like the following.
```
FATAL failed to fetch something: failed to do something else: failed to do another thing: stream:rhcos-4.8 arch:x86_64: No GCP build found
```
We are already using colons to separate different layers of the final error. The colon to separate between parts of an individual layer conflicts with that.

Reply:
Yep agreed.    The above PR merged and I've updated the vendor commit to use it.

Reply:
I'd note in *this particular* case this error really can't happen because we're fetching this data from a JSON blob compiled into the same binary; we're never going to update the installer with an RHCOS build not in GCP.  But the general error handling philosphy definitely makes sense!